TACC/Frontera-Portal
# Task/FP-866 handle potential delay in systems list retrieval

## Overview: ##

If there is a delay in getting a user's systems ([FP-866](https://jira.tacc.utexas.edu/browse/FP-866)) then we show an empty list of systems.  This PR handles those delays by:
* showing a spinner in Datafiles until the system information is received
* ensuring that allocation error message doesn't show up for an app doesn't show up because we are still waiting for systems

## Related Jira tickets: ##

* [FP-866](https://jira.tacc.utexas.edu/browse/FP-866)
* [FP-869](https://jira.tacc.utexas.edu/browse/FP-869)

## Summary of Changes: ##
* Refactor redux datafiles systems and definitions state so they have independent loading/error variables
* Handle loading/error of systems in `Datafiles` component
* Wait for `systems.datafiles.defaultHost` to be received before showing an allocation error message

## Testing Steps: ##
Test the handling of delayed systems request
1. Force systems listing to be slow by adding `import time; time.sleep(3)` to listing in `server/portal/apps/datafiles/views.py` (line 51)
2. Refresh on datafiles and ensure things work well
3. Refresh on Frontera app (`https://cep.dev/workbench/applications/frontera-hpc-jupyter-1.0u12`) and ensure that an allocation error message isn't displayed
4. (optional)Use an account missing Frontera allocation and try step 3 to make sure that the allocation error *is* displayed 

Test the handling of error in the system request
1. Force systems listing to be slow by adding 
`from portal.exceptions.api import ApiException; raise ApiException("dummy")`
 to listing in `server/portal/apps/datafiles/views.py` (line 51)
2. See the error (see screenshot below)

## UI Photos:
![Screen Shot 2021-01-27 at 7 14 40 PM](https://user-images.githubusercontent.com/8287580/106075855-0fd6ca00-60d4-11eb-9928-2e9045837939.png)

![Screen Shot 2021-01-28 at 12 09 07 PM](https://user-images.githubusercontent.com/8287580/106180480-f5e3c880-6161-11eb-8400-8316fdc0a0db.png)


## Notes: ##
TODO
- [x] Consult with Design about styling of error message in DataFiles
